### PR TITLE
@W-15280006, W-15279960 (SDB null casting), W-14444417 (SDB WeekDay function)

### PR DESCRIPTION
**Pull request by bairenlong-sfdc**
Bug fixes for 
1. [W-15280006](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001mU3vDYAS/view)
2. [W-15279960](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001mU6dwYAC/view)
3. [W-14444417](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001e0xFBYAY/view)

**All changes are noted, other diffs were done by IDE auto formatting.**

### CODE REVIEW

File: impl/src/main/java/com/force/formula/impl/sql/FormulaPostgreSQLHooks.java [hunk 1]
```diff
@@ -182,55 +182,55 @@ default String sqlAddMonths(String dateArg, Type dateArgType, String numMonths)
...
     @Override
     default String getDateFromUnixTime() {
-    	return "TO_TIMESTAMP(%s)";
+        return "TO_TIMESTAMP(%s)";
     }
 
     @Override
     default String sqlGetWeekday() {
-        return "1+EXTRACT (DOW FROM %s)::numeric";
+        return "(1+EXTRACT (DOW FROM %s)::numeric)";
```
Comment (bairenlong-sfdc) [inline impl/src/main/java/com/force/formula/impl/sql/FormulaPostgreSQLHooks.java:216]: Fix [W-14444417](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001e0xFBYAY/view) bug.

File: impl/src/main/java/com/force/formula/commands/FunctionDatetimeValue.java [hunk 2]
```diff
@@ -64,7 +64,7 @@ public SQLPair getSQL(FormulaAST node, FormulaContext context, String[] args, St
                 } else {
                     // we know it's false
                     guard = SQLPair.generateGuard(guards, "0=0");
-                    sql = "NULL";
+                    sql = getSqlHooks(context).sqlNullCast("NULL", FormulaDateTime.class);
```
Comment (bairenlong-sfdc) [inline impl/src/main/java/com/force/formula/commands/FunctionDatetimeValue.java:67]: Fix [W-15279960](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001mU6dwYAC/view) bug.

File: impl/src/main/java/com/force/formula/commands/FunctionTimeValue.java [hunk 3]
```diff
@@ -74,15 +75,15 @@ else if (inputDataType == FormulaDateTime.class) {
                 } else {
                     // we know it's false
                     guard = SQLPair.generateGuard(guards, "0=0");
-                    sql = "NULL";
+                    sql = getSqlHooks(context).sqlNullCast("NULL", FormulaTime.class);
```
Comment (bairenlong-sfdc) [inline impl/src/main/java/com/force/formula/commands/FunctionTimeValue.java:78]: Fix [W-15279960](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001mU6dwYAC/view) bug.

File: impl/src/main/java/com/force/formula/commands/FunctionDay.java [hunk 4]
```diff
@@ -38,9 +38,10 @@ public FormulaCommand getCommand(FormulaAST node, FormulaContext context) {
 
     @Override
     public SQLPair getSQL(FormulaAST node, FormulaContext context, String[] args, String[] guards, TableAliasRegistry registry) {
-        return new SQLPair(String.format(getSqlHooks(context).sqlChronoUnit(ChronoUnit.DAYS, Date.class), args[0]), guards[0]);
```
Comment (bairenlong-sfdc) [inline impl/src/main/java/com/force/formula/commands/FunctionDay.java:41]: Fix [W-15280006](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001mU3vDYAS/view) bug.

File: impl/src/main/java/com/force/formula/commands/FunctionDayOfYear.java [hunk 5]
```diff
@@ -30,32 +37,33 @@ public FormulaCommand getCommand(FormulaAST node, FormulaContext context) {
 
     @Override
     public SQLPair getSQL(FormulaAST node, FormulaContext context, String[] args, String[] guards, TableAliasRegistry registry) {
-        return new SQLPair(String.format(getSqlHooks(context).sqlGetDayOfYear(), args[0]), guards[0]);
```
Comment (bairenlong-sfdc) [inline impl/src/main/java/com/force/formula/commands/FunctionDayOfYear.java:33]: Fix [W-15280006](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001mU3vDYAS/view) bug.

File: impl/src/main/java/com/force/formula/commands/FunctionHour.java [hunk 6]
```diff
@@ -38,9 +38,10 @@ public FormulaCommand getCommand(FormulaAST node, FormulaContext context) {
 
     @Override
     public SQLPair getSQL(FormulaAST node, FormulaContext context, String[] args, String[] guards, TableAliasRegistry registry) {
-        return new SQLPair(String.format(getSqlHooks(context).sqlChronoUnit(ChronoUnit.HOURS, FormulaTime.class), args[0]), guards[0]);
```
Comment (bairenlong-sfdc) [inline impl/src/main/java/com/force/formula/commands/FunctionHour.java:41]: Fix [W-15280006](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001mU3vDYAS/view) bug.

File: impl/src/main/java/com/force/formula/commands/FunctionIsoWeek.java [hunk 7]
```diff
@@ -43,16 +43,17 @@ public FormulaCommand getCommand(FormulaAST node, FormulaContext context) {
 
     @Override
     public SQLPair getSQL(FormulaAST node, FormulaContext context, String[] args, String[] guards, TableAliasRegistry registry) {
-        return new SQLPair(String.format(getSqlHooks(context).sqlGetIsoWeek(), args[0]), guards[0]);
```
Comment (bairenlong-sfdc) [inline impl/src/main/java/com/force/formula/commands/FunctionIsoWeek.java:46]: Fix [W-15280006](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001mU3vDYAS/view) bug.

File: impl/src/main/java/com/force/formula/commands/FunctionIsoYear.java [hunk 8]
```diff
@@ -43,17 +43,18 @@ public FormulaCommand getCommand(FormulaAST node, FormulaContext context) {
 
     @Override
     public SQLPair getSQL(FormulaAST node, FormulaContext context, String[] args, String[] guards, TableAliasRegistry registry) {
-    	// Pass in the date twice because sqlserver uses is
-        return new SQLPair(String.format(getSqlHooks(context).sqlGetIsoYear(), args[0], args[0]), guards[0]);
```
Comment (bairenlong-sfdc) [inline impl/src/main/java/com/force/formula/commands/FunctionIsoYear.java:47]: Fix [W-15280006](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001mU3vDYAS/view) bug.

File: impl/src/main/java/com/force/formula/commands/FunctionMillisecond.java [hunk 9]
```diff
@@ -38,7 +38,8 @@ public FormulaCommand getCommand(FormulaAST node, FormulaContext context) {
 
     @Override
     public SQLPair getSQL(FormulaAST node, FormulaContext context, String[] args, String[] guards, TableAliasRegistry registry) {
-        return new SQLPair(String.format(getSqlHooks(context).sqlChronoUnit(ChronoUnit.MILLIS, FormulaTime.class), args[0]), guards[0]);
```
Comment (bairenlong-sfdc) [inline impl/src/main/java/com/force/formula/commands/FunctionMillisecond.java:41]: Fix [W-15280006](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001mU3vDYAS/view) bug.

File: impl/src/main/java/com/force/formula/commands/FunctionMonth.java [hunk 10]
```diff
@@ -38,9 +38,10 @@ public FormulaCommand getCommand(FormulaAST node, FormulaContext context) {
 
     @Override
     public SQLPair getSQL(FormulaAST node, FormulaContext context, String[] args, String[] guards, TableAliasRegistry registry) {
-        return new SQLPair(String.format(getSqlHooks(context).sqlChronoUnit(ChronoUnit.MONTHS, Date.class), args[0]), guards[0]);
```
Comment (bairenlong-sfdc) [inline impl/src/main/java/com/force/formula/commands/FunctionMonth.java:41]: Fix [W-15280006](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001mU3vDYAS/view) bug.

File: impl/src/main/java/com/force/formula/commands/FunctionSecond.java [hunk 11]
```diff
@@ -40,16 +40,17 @@ public FormulaCommand getCommand(FormulaAST node, FormulaContext context) {
 
     @Override
     public SQLPair getSQL(FormulaAST node, FormulaContext context, String[] args, String[] guards, TableAliasRegistry registry) {
-        return new SQLPair(String.format(getSqlHooks(context).sqlChronoUnit(ChronoUnit.SECONDS, FormulaTime.class), args[0]), guards[0]);
```
Comment (bairenlong-sfdc) [inline impl/src/main/java/com/force/formula/commands/FunctionSecond.java:43]: Fix [W-15280006](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001mU3vDYAS/view) bug.

File: impl/src/main/java/com/force/formula/commands/FunctionTimeValue.java [hunk 12]
```diff
@@ -60,9 +60,10 @@ public SQLPair getSQL(FormulaAST node, FormulaContext context, String[] args, St
             guard = SQLPair.generateGuard(guards, null);
         }
         else if (inputDataType == FormulaDateTime.class) {
-            sql = getSqlHooks(context).sqlExtractTimeFromDateTime(args[0]);
```
Comment (bairenlong-sfdc) [inline impl/src/main/java/com/force/formula/commands/FunctionTimeValue.java:63]: Fix [W-15280006](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001mU3vDYAS/view) bug.

File: impl/src/main/java/com/force/formula/commands/FunctionUnixTimestamp.java [hunk 13]
```diff
@@ -44,19 +44,21 @@ public FormulaCommand getCommand(FormulaAST node, FormulaContext context) {
 
     @Override
     public SQLPair getSQL(FormulaAST node, FormulaContext context, String[] args, String[] guards, TableAliasRegistry registry) {
-    	Type type = ((FormulaAST)node.getFirstChild()).getDataType();
```
Comment (bairenlong-sfdc) [inline impl/src/main/java/com/force/formula/commands/FunctionUnixTimestamp.java:47]: Fix [W-15280006](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001mU3vDYAS/view) bug.

File: impl/src/main/java/com/force/formula/commands/FunctionWeekday.java [hunk 14]
```diff
@@ -39,19 +39,20 @@ public FormulaCommand getCommand(FormulaAST node, FormulaContext context) {
 
     @Override
     public SQLPair getSQL(FormulaAST node, FormulaContext context, String[] args, String[] guards, TableAliasRegistry registry) {
-    	String sql = String.format( getSqlHooks(context).sqlGetWeekday(), args[0]);
```
Comment (bairenlong-sfdc) [inline impl/src/main/java/com/force/formula/commands/FunctionWeekday.java:42]: Fix [W-15280006](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001mU3vDYAS/view) bug.

File: impl/src/main/java/com/force/formula/impl/FormulaSqlHooks.java [hunk 15]
```diff
@@ -852,4 +852,13 @@ default String sqlRpad(String str, String amount, String pad) {
     default String sqlLike(String str, String like) {
         return str + " LIKE " + like + " ESCAPE '\\'";
     }
+
```
Comment (bairenlong-sfdc) [inline impl/src/main/java/com/force/formula/impl/FormulaSqlHooks.java:855]: formula-engine-sfdc/java/src/com/force/formula/sfdc/impl/FormulaSdbHooks.java will override for SDB.

File: impl/src/main/java/com/force/formula/commands/FunctionDateValue.java [hunk 16]
```diff
@@ -78,15 +78,15 @@ public SQLPair getSQL(FormulaAST node, FormulaContext context, String[] args, St
                 } else {
                     // we know it's false
                     guard = SQLPair.generateGuard(guards, "0=0");
-                    sql = String.format(getSqlHooks(context).sqlToDate(Date.class), "NULL");
```
Comment (bairenlong-sfdc) [inline impl/src/main/java/com/force/formula/commands/FunctionDateValue.java:81]: No functional changes. Make it consistent with FunctionDatetimeValue & FunctionTimeValue.

**Comment by j-yang-sfdc** [comment]
Test

File: impl/src/main/java/com/force/formula/commands/FunctionDateValue.java (hunk 16 above)
Comment (j-yang-sfdc) [reply impl/src/main/java/com/force/formula/commands/FunctionDateValue.java:81]: Test

**Review by jfreschl — APPROVED** [review]

**Pull request merged**